FBSimulatorControl/Session/FBSimDeviceWrapper.m

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "FBSimDeviceWrapper.h"
+
+#import <CoreSimulator/SimDevice.h>
+
+#import "FBSimulatorError.h"
+
+const CFTimeInterval kFBSimDeviceCommandTimeout = 30.0;
+
+@interface FBSimDeviceWrapper ()
+
+@property SimDevice *device;
+@property NSInteger lastCallID;
+
+@end
+
+@implementation FBSimDeviceWrapper
+
+- (instancetype)initWithSimDevice:(SimDevice *)device
+{
+  if (!(self = [self init])) {
+    return nil;
+  }
+
+  _device = device;
+  _lastCallID = 0;
+
+  return self;
+}
+
+- (void)scheduleInvocationInBackground:(NSInvocation *)invocation
+{
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [NSThread detachNewThreadSelector:@selector(runInvocation:) toTarget:self withObject:invocation];
```

Comment:
GCD would mean creating new queues and destroying them since we expect the call can get stuck forever. With NSThread this is all I have to do to create a new thread